poet/src/build_authors.rs

```diff
@@ -0,0 +1,46 @@
+use std::collections::BTreeMap;
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use anyhow::Result;
+use anyhow::anyhow;
+
+use crate::author::Author;
+use crate::author_basename::AuthorBasename;
+use crate::author_data::AuthorData;
+use crate::document_error_collection::DocumentErrorCollection;
+use crate::filesystem::Filesystem;
+use crate::filesystem::storage::Storage;
+
+pub async fn build_authors(
+    source_filesystem: Arc<Storage>,
+) -> Result<BTreeMap<AuthorBasename, Author>> {
```

Comment:
It could return some `AuthorCollection` instead with built-in method for filtering and validation. Could make it simpler, because passing around `BTreeMap` makes you implement helper functions all over the place (some in `render_document`, some in `build_project` etc instead). They can be contained in some custom collection wrapper.